### PR TITLE
Add e2e test for visual prompting

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes:
- [x] e2e test
- [x] enable `pip install -e .[visual_prompting]`
- [x] enable `tox devenv venv/otx -e tests-visprompt-py310`
- [x] remove unused configs
- [x] fix about no returning the first label in `VisualPromptingDatasetAdapter`

#### Issues
- [x] Due to using small samples (car_tree_bug), statistics seem unstable and occurred `data.size() == OC` during `otx optimize`. Solve by limiting activation range.
    → Local e2e was passed, but WGISD occurred the error `data.size() == OC`. PTQ for visual prompting looks like unstable, so it is temporarily skipped just for now.

|  | f32 | u8 (image encoder only) |
| -- | -- | -- |
| car_tree_bug | 0.8329 | 0.8406 |
| Trashcan | 0.8147 | 0.8172 |
| FLARE22 | 0.8951 | 0.8883 |
| WGISD | 0.9425 | `error` |

- [ ] `deterministic=True` in tests will be implemented (https://github.com/openvinotoolkit/training_extensions/pull/2361)

### How to test
```shell
# run e2e
$ tox -vv -e tests-all-py310 -- tests/e2e/cli/visual_prompting/test_visual_prompting.py
======================================================== 7 passed, 1 warning in 171.07s (0:02:51) =========================================================

# pip install -e .[visual_prompting]
$ python -m venv venv
$ source venv/bin/activate
$ (venv) pip install -e .[visual_prompting]
$ (venv) pytest tests/e2e/cli/visual_prompting/test_visual_prompting.py
======================================================== 7 passed, 1 warning in 153.34s (0:02:33) =========================================================

# tox devenv venv/otx -e tests-visprompt-py310
$ tox devenv venv/otx -e tests-visprompt-py310
$ source venv/otx/bin/activate
$ tox -vv -e tests-visprompt-py310 -- tests/e2e/cli/visual_prompting/test_visual_prompting.py
======================================================== 7 passed, 1 warning in 168.64s (0:02:48) ========================================================

# integration test
$ tox -e tests-visprompt-py310 -- tests/integration/cli/visual_prompting
==================================================== 8 passed, 2 skipped, 1 warning in 93.69s (0:01:33) ====================================================
```

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by eunwoosh — DISMISSED** [review]

**Comment by jaegukhyun** [comment]
@yunchu Could you check workflow and tox.ini changes?

File: tests/e2e/cli/visual_prompting/test_visual_prompting.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        tmp_dir_path = tmp_dir_path / "visual_prompting"
+        otx_export_testing(template, tmp_dir_path, half_precision=True)
+
+    @e2e_pytest_component
+    @pytest.mark.skipif(TT_STABILITY_TESTS, reason="This is TT_STABILITY_TESTS")
+    @pytest.mark.parametrize("template", templates, ids=templates_ids)
+    @pytest.mark.parametrize("half_precision", [True, False])
+    def test_otx_eval_openvino(self, template, tmp_dir_path, half_precision):
+        tmp_dir_path = tmp_dir_path / "visual_prompting"
+        otx_eval_openvino_testing(template, tmp_dir_path, otx_dir, args, threshold=1.0, half_precision=half_precision)
```
Comment (jaegukhyun) [inline tests/e2e/cli/visual_prompting/test_visual_prompting.py]:
```suggestion
        otx_eval_openvino_testing(template, tmp_dir_path, otx_dir, args, threshold=0.2, half_precision=half_precision)
```

File: tests/e2e/cli/visual_prompting/test_visual_prompting.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        tmp_dir_path = tmp_dir_path / "visual_prompting"
+        otx_eval_testing(template, tmp_dir_path, otx_dir, args)
+
+    @e2e_pytest_component
+    @pytest.mark.skipif(TT_STABILITY_TESTS, reason="This is TT_STABILITY_TESTS")
+    @pytest.mark.parametrize("template", templates, ids=templates_ids)
+    def test_otx_export(self, template, tmp_dir_path):
+        tmp_dir_path = tmp_dir_path / "visual_prompting"
+        otx_export_testing(template, tmp_dir_path, False)
+
```
Comment (jaegukhyun) [inline tests/e2e/cli/visual_prompting/test_visual_prompting.py:109]: Don't we need to check onnx export tc?

**Comment by yunchu** [comment]
@sungchul2 can you confirm that the integration tests are passed as well on you local env?
I think it would be done by command `tox -e tests-vp-py310 -- tests/integration/cli/visual_prompting`

**Comment by sungchul1** [comment]
> @sungchul2 can you confirm that the integration tests are passed as well on you local env? I think it would be done by command `tox -e tests-vp-py310 -- tests/integration/cli/visual_prompting`

`vp` -> `visprompt` because `vp` seems too short.
```shell
$ tox -e tests-visprompt-py310 -- tests/integration/cli/visual_prompting
==================================================== 8 passed, 2 skipped, 1 warning in 93.69s (0:01:33) ====================================================
```

File: tests/e2e/cli/visual_prompting/test_visual_prompting.py (hunk 2 above)
Comment (sungchul1) [reply tests/e2e/cli/visual_prompting/test_visual_prompting.py:109]: I'll add it.
BTW, I'm just curious, is there any onnx export tc for other tasks?

Comment (jaegukhyun) [reply tests/e2e/cli/visual_prompting/test_visual_prompting.py:109]: Ah, sorry. It's only in integration test.

**Review by sungmanc — DISMISSED** [review]
LGTM

File: tests/e2e/cli/visual_prompting/test_visual_prompting.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        tmp_dir_path = tmp_dir_path / "visual_prompting"
+        pot_validate_fq_testing(template, tmp_dir_path, otx_dir, "visual_prompting", type(self).__name__)
+
+    @e2e_pytest_component
+    @pytest.mark.skip("PTQ for visual prompting still seems unstable, skipped.")
+    @pytest.mark.skipif(TT_STABILITY_TESTS, reason="This is TT_STABILITY_TESTS")
+    @pytest.mark.parametrize("template", templates, ids=templates_ids)
+    def test_pot_eval(self, template, tmp_dir_path):
+        tmp_dir_path = tmp_dir_path / "visual_prompting"
+        pot_eval_testing(template, tmp_dir_path, otx_dir, args)
```
Comment (jaegukhyun) [inline tests/e2e/cli/visual_prompting/test_visual_prompting.py]: I think removing this tc would be more appropriate. Let's add these tc after stabilizing ptq.

**Comment by yunchu** [comment]
@sungchul2 can you clarify that why e2e testing results for visual prompting show different number of passed/skipped in your PR description?
image

**Comment by sungchul2** [comment]
> 

Updated.

**Comment by sungchul1** [comment]
> @sungchul2 can you clarify that why e2e testing results for visual prompting show different number of passed/skipped in your PR description? image

Updated.

File: tests/e2e/cli/visual_prompting/test_visual_prompting.py (hunk 3 above)
Comment (sungchul1) [reply tests/e2e/cli/visual_prompting/test_visual_prompting.py]: Done.

**Review by yunchu — APPROVED** [review]
LGTM

**Review by sungmanc — APPROVED** [review]

**Pull request merged**